File: folkmq-lang-js/folkmq-test/demo02-node-ws/_broadcast_test.js
const {FolkMQ} = require("@noear/folkmq");
const {MqMessage} = require("@noear/folkmq/client/MqMessage");

async function main() {
    const client1 =  await FolkMQ.createClient("folkmq:ws://127.0.0.1:18602")
        .nameAs("demoapp")
        .connect();

    const client2 =  await FolkMQ.createClient("folkmq:ws://127.0.0.1:18602")
        .nameAs("demoapp")
        .connect();

    //订阅
    client1.subscribe("test", null, true, message => {
        console.log(message.getBodyAsString());
    });

    client2.subscribe("test", null, true, message => {
        console.log(message.getBodyAsString());
    });

    //发布
    client1.publish("test", new MqMessage("hot-1").broadcast(false));
    client1.publish("test", new MqMessage("hot-2").broadcast(true));
}

main();

File: folkmq-lang-python/folkmq/folkmq-test/demo02/_broadcast_test.py
import asyncio

from socketd.utils.LogConfig import log

from folkmq.FolkMQ import FolkMQ


async def main():
    client1 = await (FolkMQ.create_client("folkmq:ws://127.0.0.1:18602")
                    .name_as("demoapp")
                    .connect())
    client2 = await (FolkMQ.create_client("folkmq:ws://127.0.0.1:18602")
                     .name_as("demoapp")
                     .connect())

    # 订阅
    await client1.subscribe("test", None, True, lambda message:
        log.info(message.get_body_as_string())
    )
    await client2.subscribe("test", None, True, lambda message:
        log.info(message.get_body_as_string())
    )

    # 发布
    await client1.publish("test", FolkMQ.new_message("hot-1"))
    await client1.publish("test", FolkMQ.new_message("hot-2").broadcast(True))

    await asyncio.Future()

if __name__ == "__main__":
    asyncio.run(main())

